iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/LineAdderAdapter.java

```diff
@@ -52,24 +57,83 @@
 
     private double b2 = Double.NaN;
 
+    private double p0 = Double.NaN;
+
+    private double q0 = Double.NaN;
+
+    private String ucteXnodeCode;
+
     LineAdderAdapter(final MergingViewIndex index) {
         this.index = Objects.requireNonNull(index, "merging view index is null");
     }
 
     @Override
     public Line add() {
-        Line newLine = null;
+        Line newLine;
         final Network n1 = checkAndGetNetwork1();
         final Network n2 = checkAndGetNetwork2();
         if (n1 == n2) {
             newLine = index.getLine(addLine(n1));
         } else {
+            // FIXME : how to compute P0, Q0 & ucteXnodeCode ?
+            ucteXnodeCode = "";
+            p0 = 0.0;
+            q0 = 0.0;
```

Comment:
The P0 and Q0 are equal to the half of the sum of P1 and P2 (respectively Q1 and Q2). The sign depends on the side.
If P1 and P2 are NaN, you cannot initialize these values. You probably have to listen (NetworkListener) an event and recompute them each time P1, P2, Q1, Q2 are updated.

Reply:
So, I keep Line creation with q0 & p0 equal to 0.0 (NaN are not allowed)
I implement NetworkListener::onUpdate with variantId into MergingNetworkListener and compute P0 & Q0 if P1, P2, Q1 or Q2 are updated.

How i set ucteXnodeCode ?

Reply:
@tadam50 The xnode code is not a mandatory information so you can just skip it. The idea we have is to use both Xnode and MergedXnode extensions and initialize the attributes from these extensions, so not directly when the object is created.
As the name of the XNode is not mandatory, you have to find another way to cache the merged lines

About P and Q, the solution I suggested is ok.

Reply:
Issue #1183 created to manage Xnode